device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/MultiplexingClient.java

```diff
@@ -200,13 +244,100 @@ public void close() throws IOException
+     */
+    public void registerDeviceClient(DeviceClient deviceClient, long timeoutMilliseconds) throws InterruptedException, MultiplexingClientException {
         Objects.requireNonNull(deviceClient);
         List<DeviceClient> clientList = new ArrayList<>();
         clientList.add(deviceClient);
-        registerDeviceClients(clientList);
+        registerDeviceClients(clientList, timeoutMilliseconds);
+    }
+
+    /**
+     * Register device clients to this multiplexing client. This method may be called before or after opening the multiplexed
+     * connection.
+     * <p>
+     * Up to {@link #MAX_MULTIPLEX_DEVICE_COUNT_AMQPS} devices can be registered on a multiplexed AMQPS connection,
+     * and up to {@link #MAX_MULTIPLEX_DEVICE_COUNT_AMQPS_WS} devices can be registered on a multiplexed AMQPS_WS connection.
+     * <p>
+     * If the multiplexing client is already open, then these device clients will automatically
+     * be opened, too. If the multiplexing client is not open yet, then these device clients will not be opened until
+     * {@link MultiplexingClient#open()} is called.
+     * <p>
+     * If the multiplexed connection is already open, then this call will asynchronously add each device client to the
+     * multiplexed connection, and then will block until all registrations have been completed.
+     * <p>
+     * Any proxy settings set to the provided device clients will be overwritten by the proxy settings of this multiplexing client.
+     * <p>
+     * The registered device clients must use the same transport protocol (AMQPS or AMQPS_WS) that this multiplexing client uses.
+     * <p>
+     * Each registered device client may have its own retry policy and its own SAS token expiry time, separate from every other registered device client.
+     * <p>
+     * The registered device clients must use symmetric key based authentication.
+     * <p>
+     * The registered device clients must belong to the same IoT Hub as all previously registered device clients.
+     * <p>
+     * If any of these device clients are already registered to this multiplexing client, then then this method will
+     * not do anything to that particular device client. All other provided device clients will still be registered though.
+     * <p>
+     * @throws InterruptedException If the thread gets interrupted while waiting for the registration to succeed. This
+     * will never be thrown if the multiplexing client is not open yet.
+     * @throws MultiplexingClientDeviceRegistrationFailedException If one or more devices failed to register. Details for each failure can be found
+     * in this exception. Any devices not found in the map of registration exceptions provided by this exception have registered successfully.
+     * @throws com.microsoft.azure.sdk.iot.device.exceptions.MultiplexingClientDeviceRegistrationTimeoutException If this operation takes longer than the default timeout allows.
```

Comment:
does `com.microsoft.azure.sdk.iot.device.exceptions.MultiplexingClientDeviceRegistrationTimeoutException` need to be fully qualified?

Reply:
Strangely, yes. I can't figure out why though